### PR TITLE
Product page added

### PR DESCRIPTION
**Pull request by sara-db**
- Added product page
- Added link to product page for each product listed in the product list page

### CODE REVIEW

File: diffblue.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+buildCmd: mvn compile
```
Comment (jgwilson42) [inline diffblue.yml]: Please could you rebase on the latest master as this file should not have changed with this PR.

File: src/main/resources/templates/ProductPage.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    <link rel="stylesheet" href="css/bootstrap.min.css"/>
+</head>
+<body>
+<h1>Product Page</h1>
+<p>Product id: <span th:text="${product.id}"/></p>
+<p>Product name: <span th:text="${product.name}"/></p>
+<p>Product price: <span th:text="${product.price}"/></p>
+<p>Copyright 2016-2017 DiffBlue limited. All rights reserved.</p>
+</body>
+</html>
```
Comment (jgwilson42) [inline src/main/resources/templates/ProductPage.html]: Please add a blank line at the end of this file.

**Review by jgwilson42 — CHANGES_REQUESTED** [review]
Hi Sara,

Couple of small changes. Also, please could you add to the description of the PR the functionality that is being provided by this change.

Thanks,
James

**Review by forejtv — APPROVED** [review]

File: src/main/java/com/diffblue/demo/ecommerce/ProductController.java [hunk 3]
```diff
@@ -39,4 +38,16 @@ public String productList(Map<String, Object> model) {
...
 
+  @RequestMapping("/product/{id}")
+  public String productPage(@PathVariable("id") String id, Map<String, Object> model) {
+    Product prod = this.productRepo.findById(id);
+    if(prod != null){
+      Application.log.info("Product as string: " + prod.toString());
+      model.put("product", prod);
+      return "ProductPage";
+    }else{
+      return "redirect:/ProductList";
```
Comment (jgwilson42) [inline src/main/java/com/diffblue/demo/ecommerce/ProductController.java]: Would it be helpful to log here the id that was being tried to the application log?

File: src/main/java/com/diffblue/demo/ecommerce/ProductRepository.java [hunk 4]
```diff
@@ -17,5 +17,5 @@
    */
   
   Iterable<Product> findAll();
-
+  Product findById(String id);
```
Comment (jgwilson42) [inline src/main/java/com/diffblue/demo/ecommerce/ProductRepository.java]: Please could you update the javadoc comment to give information about this?

**Review by jgwilson42 — CHANGES_REQUESTED** [review]
Couple of minor comments, also please could you squash your commits to remove the ones that are not appropriate?

File: src/main/resources/templates/ProductPage.html (hunk 2 above)
Comment (sara-db) [reply src/main/resources/templates/ProductPage.html]: Addressed.

File: src/main/java/com/diffblue/demo/ecommerce/ProductController.java (hunk 3 above)
Comment (sara-db) [reply src/main/java/com/diffblue/demo/ecommerce/ProductController.java]: Addressed.

**Review by paulkennethkent — APPROVED** [review]

**Review by jgwilson42 — CHANGES_REQUESTED** [review]
Hi Sara,

Thanks for making the changes, please could you also address the coding style comments from the audit in the build?

```
/home/travis/build/diffblue/ecommerce-demo/src/main/java/com/diffblue/demo/ecommerce/ProductController.java:8: warning: Wrong lexicographical order for 'org.springframework.web.bind.annotation.PathVariable' import. Should be before 'org.springframework.web.bind.annotation.RequestMapping'.
/home/travis/build/diffblue/ecommerce-demo/src/main/java/com/diffblue/demo/ecommerce/ProductController.java:9: warning: 'java.util.Map' should be separated from previous import group.
/home/travis/build/diffblue/ecommerce-demo/src/main/java/com/diffblue/demo/ecommerce/ProductController.java:41:3: warning: Missing a Javadoc comment.
/home/travis/build/diffblue/ecommerce-demo/src/main/java/com/diffblue/demo/ecommerce/ProductController.java:44:7: warning: WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
/home/travis/build/diffblue/ecommerce-demo/src/main/java/com/diffblue/demo/ecommerce/ProductController.java:44:21: warning: WhitespaceAround: '{' is not preceded with whitespace.
/home/travis/build/diffblue/ecommerce-demo/src/main/java/com/diffblue/demo/ecommerce/ProductController.java:48:6: warning: WhitespaceAround: 'else' is not preceded with whitespace.
/home/travis/build/diffblue/ecommerce-demo/src/main/java/com/diffblue/demo/ecommerce/ProductController.java:48:6: warning: WhitespaceAround: '}' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
/home/travis/build/diffblue/ecommerce-demo/src/main/java/com/diffblue/demo/ecommerce/ProductController.java:48:10: warning: WhitespaceAround: 'else' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
/home/travis/build/diffblue/ecommerce-demo/src/main/java/com/diffblue/demo/ecommerce/ProductController.java:48:10: warning: WhitespaceAround: '{' is not preceded with whitespace.
Audit done.
```

Thanks,
James

**Review by jgwilson42 — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]

**Pull request merged**